core/src/main/java/nl/nn/adapterframework/scheduler/job/CleanupDatabaseJob.java

```diff
@@ -98,34 +99,28 @@ public String getTypeField() {
 	}
 
 	@Override
-	public void execute(IbisManager ibisManager) {
-		Date date = new Date();
-
-		int maxRows = AppConstants.getInstance().getInt("cleanup.database.maxrows", 25000);
+	public void beforeExecuteJob(IbisManager ibisManager) {
+		Set<String> datasourceNames = getAllLockerDatasourceNames(ibisManager);
 
-		List<String> datasourceNames = getAllLockerDatasourceNames(ibisManager);
-
-		for (Iterator<String> iter = datasourceNames.iterator(); iter.hasNext();) {
-			String datasourceName = iter.next();
+		for (String datasourceName : datasourceNames) {
 			FixedQuerySender qs = null;
 			try {
 				qs = SpringUtils.createBean(getApplicationContext(), FixedQuerySender.class);
 				qs.setDatasourceName(datasourceName);
 				qs.setName("cleanupDatabase-IBISLOCK");
 				qs.setQueryType("other");
-				qs.setTimeout(getQueryTimeout());
```

Comment:
I would say it is a good idea to keep the timeout, to be able to get out of deadlock situations